src/Features/Core/Portable/CodeRefactorings/UseNamedArguments/AbstractUseNamedArgumentsCodeRefactoringProvider.cs

```diff
@@ -84,18 +84,22 @@ protected abstract class Analyzer<TBaseArgumentSyntax, TArgumentSyntax, TArgumen
                 }
 
                 var argumentList = (TArgumentListSyntax)argument.Parent;
-                var (index, count) = GetArgumentListIndexAndCount(argument, argumentList);
-
                 var arguments = GetArguments(argumentList);
-                for (var i = index; i < count; i++)
+                var index = arguments.IndexOf(argument);
+                if (index == -1 || index >= parameters.Length)
```

Comment:
can you call it argumentIndex to make it clear that it may have a different range than parameters' range.